boards/arm/sam3u_ek/Kconfig.board

```diff
@@ -0,0 +1,8 @@
+# Atmel SAM3U-EK selection
+
+# Copyright (c) 2023 Chen Xingyu <hi@xingrz.me>
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_SAM3U_EK
+	bool "Atmel SAM3U-EK"
+	depends on SOC_PART_NUMBER_SAM3U4C
```

Comment:
This commit should be split in:

1) only west.yml (it should be first in the whole series)
2) dts folder entries
3) soc folder content
4) drivers/<any>
5) board